### DISCUSSION TITLE
Construction of explicit multi-electron function with PySCF

### DISCUSSION THREAD

**Discussion by Physfock** [pyscf/pyscf#2640] [Q&A] [answered]
Hello, everyone. I am new to quantum chemistry. I have a question how to get the wave function in PySCF (single-determinant e.g. within the Hartree-fock method) as a function on a coordinate grid. For example, I need a molecule electron wave function for H2 \Psi(\vec{r}_1, \vec{r}_2). This would be the Slatter determinant of \vec{r}_1 and \vec{r}_2 (a function of six variables in total). For example:

import numpy as np
from pyscf import gto, scf, dft

mol = gto.M(
    atom = 'H 0 0 0; H 0 0 0.74',  
    basis = 'sto-3g',
    unit = 'angstrom'
)

mf = scf.RHF(mol)
mf.kernel()

grid = dft.gen_grid.Grids(mol)
grid.level = 3  
grid.build()

coords = grid.coords  
weights = grid.weights  

orbital_values = mf.mol.eval_gto('GTOval', coords) @ mf.mo_coeff[:, 0]

print(orbital_values[:10])

But how to construct \Psi(\vec{r}_1, \vec{r}_2)? Maybe there is a simple way...

**Comment by sunqm** [comment, marked as answer]
You can simply compute the determinants of these orbitals for all combinations of coordinates:
```
for i in range(ngrids):
    for j in range(ngrids):
        np.linalg.det(orbital_values[[i, j]])
```
